torch/amp/autocast_mode.py

```diff
@@ -228,53 +204,26 @@ def __init__(self, device_type : str,
-                error_message = 'In IPU autocast, but the target dtype is not supported. Disabling autocast.\n'
-                error_message += 'IPU Autocast only supports dtypes of torch.bfloat16 and torch.float16 currently.'
-                warnings.warn(error_message)
-                enabled = False
-        elif self.device == 'hpu':
-            supported_dtype = [torch.bfloat16, torch.float16]
-            if self.fast_dtype not in supported_dtype:
-                error_message = 'In HPU autocast, but the target dtype is not supported. Disabling autocast.\n'
-                error_message += 'HPU Autocast only supports dtypes of torch.bfloat16 and torch.float16 currently.'
-                warnings.warn(error_message)
-                enabled = False
-        elif self.device == self.custom_backend_name:
-            supported_dtype = self.custom_device_mod.get_amp_supported_dtype()
-            if self.fast_dtype not in supported_dtype:
-                error_message = f"In {self.custom_backend_name} autocast, but the target dtype is not supported. "
-                error_message += f"Disabling autocast.\n {self.custom_backend_name} Autocast only supports dtypes of "
-                error_message += ", ".join(str(dtype) for dtype in supported_dtype) + " currently."
-                warnings.warn(error_message)
-                enabled = False
-        elif self.device == 'cuda':
-            if enabled and self.fast_dtype == torch.bfloat16 and not torch.cuda.is_bf16_supported():
-                raise RuntimeError('Current CUDA Device does not support bfloat16. Please switch dtype to float16.')
-        elif self.device == 'xla':
-            supported_dtype = [torch.bfloat16]
-            if self.fast_dtype not in supported_dtype:
-                error_message = 'In XLA autocast, but the target dtype is not supported. Disabling autocast.\n'
-                error_message += 'XLA Autocast only supports dtype of torch.bfloat16 currently.'
-                warnings.warn(error_message)
-                enabled = False
+        self.device_supported_fast_dtype_map = {
+            "cpu": [torch.bfloat16],
+            "cuda": [torch.bfloat16, torch.float16],
+            "xpu": [torch.bfloat16, torch.float16],
+            "ipu": [torch.bfloat16, torch.float16],
+            "hpu": [torch.bfloat16, torch.float16],
+            "xla": [torch.bfloat16],
+        }
+        if self.custom_backend_name == self.device:
+            self.device_supported_fast_dtype_map[self.custom_backend_name] = \
+                getattr(torch, self.device).get_amp_supported_dtype()
+        if self.device == "cuda" and enabled and self.fast_dtype == torch.bfloat16 \
+                and not torch.cuda.is_bf16_supported():
+            raise RuntimeError('Current CUDA Device does not support bfloat16. Please switch dtype to float16.')
+        supported_dtyps = self.device_supported_fast_dtype_map[self.device]
```

Comment:
nit: supported_dtyps -> supported_dtypes

Reply:
> nit: supported_dtyps -> supported_dtypes

could you have a look again? @soulitzer